### DISCUSSION TITLE
Using Calculate Formula Reactively

### DISCUSSION THREAD

**Discussion by BrianHung** [handsontable/hyperformula#1152] [API Questions]
Is there a way to use [`calculateFormula`](https://hyperformula.handsontable.com/api/classes/hyperformula.html#calculateformula) reactively?

For example, I want to build a selection summary that applies a formula like `SUM` across all cells in a selection range. I can convert that selection range to use `A1` notation, e.g. `=SUM(A1:D10)`.

But am wondering how to update that result whenever any value within `A1:D10` changes. In this context, I sort of want the normal reactivity as if `=SUM(Grid1!A1:D10)` was a cell but _not_ have hack it into a sheet.

The ideal DX I'm looking for is something like this so I can agnostically integrate it into React.
```ts
const unsubscribe = subscribe(`=SUM(Grid1!A1:D10)`, newValue => console.log(newValue));
```

I have tried looking at handsontable's [`ColumnSummary` plugin](https://github.com/handsontable/handsontable/blob/develop/handsontable/src/plugins/columnSummary/) as inspiration, and it feels close?

Just need a utility function to parse a generic formula for all of its cell dependents, then check if changes from the `valuesUpdated` listener intersect the dependents.

**Comment by AMBudnik** [comment]
Hi @BrianHung, thanks for reaching out!

I think the best approach to this problem would be to define a named expression and use the [valuesUpdated](https://hyperformula.handsontable.com/api/interfaces/listeners.html#valuesupdated) event to listen for changes in the named expression value.

demo: https://stackblitz.com/edit/js-4kvpek?file=index.html,index.js

**Reply by BrianHung** [reply]
Oh! I didn't know that `valuesUpdated` contained changes for `namedExpression`. Works perfectly for this use case.

**Reply by AMBudnik** [reply]
I'm more than happy to hear that it meets your requirements.
